### PR TITLE
use format option and raise error if format is invalid

### PR DESCRIPTION
**Pull request by ThePeiggy**

### CODE REVIEW

File: lib/pheme/queue_poller.rb [hunk 1]
```diff
@@ -36,12 +36,15 @@ def poll
...
+      case format
+      when :csv
         parsed_body = SmarterCSV.process(StringIO.new(body['Message']))
         parsed_body.map{ |item| RecursiveOpenStruct.new(item, recurse_over_arrays: true) }
-      else
+      when :json
         parsed_body = JSON.parse(body['Message'])
         RecursiveOpenStruct.new(parsed_body, recurse_over_arrays: true)
+      else
+        raise "Error: invalid message format"
```
Comment (6) [inline lib/pheme/queue_poller.rb]: could we instead use a more explicit error type, e.g.:

```ruby
raise ArgumentError.new("Invalid format")
```

File: lib/pheme/queue_poller.rb [hunk 2]
```diff
@@ -36,12 +36,15 @@ def poll
     def parse_message(message)
       Pheme.log(:info, "Received JSON payload: #{message.body}")
       body = JSON.parse(message.body)
-      if csv
+      case format
+      when :csv
         parsed_body = SmarterCSV.process(StringIO.new(body['Message']))
         parsed_body.map{ |item| RecursiveOpenStruct.new(item, recurse_over_arrays: true) }
-      else
```
Comment (cabello) [inline lib/pheme/queue_poller.rb]: Extract this to a private method `parse_csv`, and the one below `parse_json`

**Comment by cabello** [comment]
I  ❤️  this.

**Comment by ThePeiggy** [comment]
done!

**Comment by cabello** [comment]
🚀 🇮🇹

**Pull request merged**